wagtail/wagtailadmin/views/home.py

```diff
@@ -82,8 +83,16 @@ def home(request):
     for fn in hooks.get_hooks('construct_homepage_panels'):
         fn(request, panels)
 
+    root_page = get_explorable_root_page(request.user)
+    if root_page:
+        root_site = root_page.get_site()
+    else:
+        root_site = None
+
     return render(request, "wagtailadmin/home.html", {
-        'site_name': settings.WAGTAIL_SITE_NAME,
+        'root_page': root_page,
+        'root_site': root_site,
+        'site_name': root_site.site_name if root_site is not None else settings.WAGTAIL_SITE_NAME,
```

Comment:
The `site_name` field of Site is optional (and blank in the default install), so we should check for this being blank and fall back on `root_site.hostname` if so (or possibly fall back directly to `WAGTAIL_SITE_NAME` - but I think the hostname makes more sense).